lib/xcodeproj/scheme.rb

```diff
@@ -119,33 +176,14 @@ def add_test_target(test_target)
     #        A target used by scheme in the launch step.
     #
     def set_launch_target(build_target)
-      launch_product_runnable = @launch_action.add_element 'BuildableProductRunnable'
-      launch_product_runnable.attributes['runnableDebuggingMode'] = '0'
-
-      launch_buildable_reference = launch_product_runnable.add_element 'BuildableReference'
-      launch_buildable_reference.attributes['BuildableIdentifier'] = 'primary'
-      launch_buildable_reference.attributes['BlueprintIdentifier'] = build_target.uuid
-      launch_buildable_reference.attributes['BuildableName'] = "#{build_target.name}.app"
-      launch_buildable_reference.attributes['BlueprintName'] = build_target.name
-      launch_buildable_reference.attributes['ReferencedContainer'] = construct_referenced_container_uri(build_target)
-
-      profile_product_runnable = @profile_action.add_element 'BuildableProductRunnable'
-
-      profile_buildable_reference = profile_product_runnable.add_element 'BuildableReference'
-      profile_buildable_reference.attributes['BuildableIdentifier'] = 'primary'
-      profile_buildable_reference.attributes['BlueprintIdentifier'] = build_target.uuid
-      profile_buildable_reference.attributes['BuildableName'] = "#{build_target.name}.app"
-      profile_buildable_reference.attributes['BlueprintName'] = build_target.name
-      profile_buildable_reference.attributes['ReferencedContainer'] = construct_referenced_container_uri(build_target)
-
-      macro_expansion = @test_action.add_element 'MacroExpansion'
-
-      buildable_reference = macro_expansion.add_element 'BuildableReference'
-      buildable_reference.attributes['BuildableIdentifier'] = 'primary'
-      buildable_reference.attributes['BlueprintIdentifier'] = build_target.uuid
-      buildable_reference.attributes['BuildableName'] = File.basename(build_target.product_reference.path)
-      buildable_reference.attributes['BlueprintName'] = build_target.name
-      buildable_reference.attributes['ReferencedContainer'] = construct_referenced_container_uri(build_target)
+      launch_runnable = BuildableProductRunnable.new(build_target, 0)
+      launch_action.build_product_runnable = launch_runnable
+
+      profile_runnable = BuildableProductRunnable.new(build_target)
+      profile_action.build_product_runnable = profile_runnable
+
+      macro_exp = MacroExpansion.new(build_target)
+      test_action.add_macro_expansion(macro_exp)
```

Comment:
If anyone knows what is the purpose of this `<MacroExpansion>` tag in the `xcscheme`'s XML (what does it affect in practice, I'm interested (I don't see any related setting in Xcode UI?)